src/app/data_harvester/processes/windows.rs

```diff
@@ -42,11 +41,11 @@ pub fn get_process_data(
         };
 
         let pcu = {
-            let p = process_val.cpu_usage() as f64 / num_processors;
-            if p.is_nan() {
-                process_val.cpu_usage() as f64
-            } else {
+            let p = process_val.cpu_usage() as f64;
```

Comment:
The `p.is_nan()` check is mostly in the case of sysinfo reporting zero cores, which would cause a divide by zero and therefore saving a `NaN`. Since you've moved the division, the logic should now be like (apologies for the messy code):

```rust
let p = process_val.cpu_usage() as f64;
if per_core_percentage {
  let res = p / (sys.cpus().len() as f64);

  // Make sure the division didn't fail...
  if res.is_nan() {
    p
  } else {
    res
  }
} else {
  p
}
```

Alternatively, you can just check if `sys.cpus().len()` is zero beforehand.

Reply:
Turns out that my original code here is also wrong for this intention - seems like a division by zero would return `inf`. So it would be good to either change this to `is_normal()` (I'll probably patch this in master) or check if the divisor is a zero beforehand.

Reply:
Rebased and addressed in the new commit.